algorithms/linfa-elasticnet/src/algorithm.rs

```diff
@@ -130,6 +215,66 @@ impl<F: Float> ElasticNet<F> {
     }
 }
 
+/// View the fitted parameters and make predictions with a fitted
+/// elastic net model
+impl<F: Float> MultiTaskElasticNet<F> {
+    /// Get the fitted hyperplane
+    pub fn hyperplane(&self) -> &Array2<F> {
+        &self.hyperplane
+    }
+
+    /// Get the fitted intercept, [0., ..., 0.] if no intercept was fitted
+    /// Note that there are as many intercepts as tasks
+    pub fn intercept(&self) -> &Array1<F> {
+        &self.intercept
+    }
+
+    /// Get the number of steps taken in optimization algorithm
+    pub fn n_steps(&self) -> u32 {
+        self.n_steps
+    }
+
+    /// Get the duality gap at the end of the optimization algorithm
+    pub fn duality_gap(&self) -> F {
+        self.duality_gap
+    }
+
+    /// Calculate the Z score
+    pub fn z_score(&self) -> Result<Array2<F>> {
+        self.variance
+            .as_ref()
+            .map(|variance| {
+                self.hyperplane
+                    .axis_iter(Axis(0))
+                    .zip(variance.iter())
+                    .map(|(a, b)| a.to_owned() / b.sqrt())
+                    .collect()
```

Comment:
@YuhanLiin My compiler is not happy when I try to build a 2d array from an iterator of 1d arrays: `a value of type ArrayBase<OwnedRepr<F>, Dim<[usize; 2]>> cannot be built from an iterator over elements of type ArrayBase<OwnedRepr<F>, Dim<[usize; 1]>>` . Do you know how to circumvent this issue?

Reply:
The typical iterator `map().collect()` pattern doesn't work on 2D arrays. Try `Zip::from(self.hyperplane).and_broadcast(variance).map_collect(|a, b| a / b.sqrt())`